.github/actions/milestone-it/entrypoint.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/sh
+set -e
+
+# 1. Read current version.
+
+version=$(jq -r '.version' package.json)
+
+IFS='.' read -ra parts <<< "$version"
+major=${parts[0]}
+minor=${parts[1]}
+
+# 2. Determine next milestone.
+
+if [ minor == '9' ]; then
+	major=$((major+1))
+else
+	minor=$((minor+1))
+fi
+
+milestone="Gutenberg $major.$minor"
+
+# 3. Create milestone. This may fail for duplicates, which is expected and
+#    ignored.
+
+curl \
+	--silent \
+	-X POST \
+	-H "Authorization: token $GITHUB_TOKEN" \
+	-H "Content-Type: application/json" \
+	-d "{\"title\":\"$milestone\"}" \
+	"https://api.github.com/repos/$GITHUB_REPOSITORY/milestones" > /dev/null
+
+# 4. Find milestone number. This could be improved to allow for non-open status
+#    or paginated results.
+
+number=$(
+	curl \
+		--silent \
+		-H "Authorization: token $GITHUB_TOKEN" \
+		"https://api.github.com/repos/$GITHUB_REPOSITORY/milestones" \
+		| jq ".[0] | select(.title == \"$milestone\") | .number"
+)
```

Comment:
It feels like this call could be doable before the previous one to avoid the duplicate error? and the previous does return the number of the created milestone if needed? Not sure if it's better as I understand the current implementation might be simpler.

Reply:
> It feels like this call could be doable before the previous one to avoid the duplicate error? and the previous does return the number of the created milestone if needed? Not sure if it's better as I understand the current implementation might be simpler.

Maybe. To me, sure, it's a failed response, but it doesn't really matter, and it keeps the implementation a bit simpler.

For context, I implemented it originally as a way to avoid fetching milestones in the repository, since I knew the results could be paginated (at least when including "all" items). This was before I realized I needed the numeric ID of the milestone to be able to update the issue, at which point I added back the fetch.